sm/plugin_sm/src/plugin.rs

```diff
@@ -237,4 +261,35 @@ impl Plugin for ExternalPluginCommand {
             })
         }
     }
+
+    async fn download(&self, module: &SoftwareModule) -> Result<PathBuf, SoftwareError> {
+        let mut filename = module.name.to_string();
+        if let Some(version) = &module.version {
+            filename.push('_');
+            filename.push_str(version.as_str());
+        }
+        // ? This doesn't seem like the correct place to add file extension, maybe this should be left out and plugin should handle it?
+        filename.push_str(".deb");
```

Comment:
This is going to go away from here as this either has to be added on the plugin level if required (`apt-get` needs to have `.deb` in the name, but dpkg works ok without it).

It could be an option to derive it from url or request response but it is not straightforward as there are different schemes to support it, eg:
c8y - has a header with full filename
deb-archive - uses full filename in the url

So I think it may be safer to delegate it to the plugin rather than derive from url or request.

Reply:
I agree this is safer/simpler to delegate this to the plugin. Only the plugin is aware of any naming restriction and it can easily rename the file after its own requirements.